src/chart/bar/BarView.js

```diff
@@ -127,15 +128,30 @@ export default echarts.extendChartView({
 
         var roundCap = seriesModel.get('roundCap', true);
 
+        var drawBackground = seriesModel.get('showBackground');
+        var backgroundModel = seriesModel.getModel('backgroundStyle');
+
+        var bgEls = [];
+        var oldBgEls = this._backgroundEls || [];
+
+        var oldLargeBackgroundEl = group.childOfName('largeBackground');
+        group.remove(oldLargeBackgroundEl);
+
         data.diff(oldData)
             .add(function (dataIndex) {
+                var itemModel = data.getItemModel(dataIndex);
+                var layout = getLayout[coord.type](data, dataIndex, itemModel);
+
+                if (drawBackground) {
+                    var bgEl = getBackgroundEl(coord, isHorizontalOrRadial, layout);
+                    bgEl.useStyle(backgroundModel.getBarItemStyle());
+                    bgEls.push(bgEl);
```

Comment:
Should better be `bgEls[dataIndex] = bgEl` in case that the callback of `add` and `update` are called in a different order from the param `dataIndex`.